pkg/solana/chainwriter/chain_writer_test.go

```diff
@@ -0,0 +1,692 @@
+package chainwriter_test
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"reflect"
+	"testing"
+
+	ag_binary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+	addresslookuptable "github.com/gagliardetto/solana-go/programs/address-lookup-table"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/monitoring/testutils"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/chainwriter"
+	clientmocks "github.com/smartcontractkit/chainlink-solana/pkg/solana/client/mocks"
+	feemocks "github.com/smartcontractkit/chainlink-solana/pkg/solana/fees/mocks"
+	txmMocks "github.com/smartcontractkit/chainlink-solana/pkg/solana/txm/mocks"
+)
+
+var testContractIDLJson = `{"version":"0.1.0","name":"contractReaderInterface","instructions":[{"name":"initialize","accounts":[{"name":"data","isMut":true,"isSigner":false},{"name":"signer","isMut":true,"isSigner":true},{"name":"systemProgram","isMut":false,"isSigner":false}],"args":[{"name":"testIdx","type":"u64"},{"name":"value","type":"u64"}]},{"name":"initializeLookupTable","accounts":[{"name":"writeDataAccount","isMut":true,"isSigner":false,"docs":["PDA for LookupTableDataAccount, derived from seeds and created by the System Program"]},{"name":"admin","isMut":true,"isSigner":true,"docs":["Admin account that pays for PDA creation and signs the transaction"]},{"name":"systemProgram","isMut":false,"isSigner":false,"docs":["System Program required for PDA creation"]}],"args":[{"name":"lookupTable","type":"publicKey"}]}],"accounts":[{"name":"LookupTableDataAccount","type":{"kind":"struct","fields":[{"name":"version","type":"u8"},{"name":"administrator","type":"publicKey"},{"name":"pendingAdministrator","type":"publicKey"},{"name":"lookupTable","type":"publicKey"}]}},{"name":"DataAccount","type":{"kind":"struct","fields":[{"name":"idx","type":"u64"},{"name":"bump","type":"u8"},{"name":"u64Value","type":"u64"},{"name":"u64Slice","type":{"vec":"u64"}}]}}]}`
```

Comment:
nit/optional: move into a new json file with better formatting, so easier to read this.

For reference, see this example: https://github.com/smartcontractkit/chainlink-internal-integrations/blob/bfb5305929ea0a6f81924d3181a898b5e8e97a7c/solana/pkg/solana/codec/testutils/testIDL.json#L13

But I understand, this PR is huge, and ok to let it be like this right now. Maybe we can do this work later.

Reply:
Added a TODO note with a ticket linked for this https://github.com/smartcontractkit/chainlink-solana/blob/0e212288f68b2b5fe17afa47867645f9f9eeee02/pkg/solana/chainwriter/chain_writer_test.go#L28

Reply:
I didn't mean to say move it to the different repo. Just to move it to a local json file, so it helps in readability.
Also fixed the ticket

Reply:
Ah I see. I just went ahead and fixed that